frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -29,17 +36,130 @@ export type VolumeSnapshotModalProps = {
+              href="https://en.wikipedia.org/wiki/Cron#Overview"
+            >
+              Whats this?
+            </a>
+          </label>
+          <TextInput
+            isRequired
+            type="text"
+            name="snapshot-modal__schedule"
+            aria-label="snapshot-modal__schedule"
+            value={scheduleLabel}
+            onChange={(value) => setScheduleLabel(value)}
+            className="ceph-volume-snapshot-modal--label"
+            placeholder="**5***"
+          />
+          <span className="help-block">* Scheduled for UTC+3</span>
+        </FormGroup>
+        <FormGroup
+          className="ceph-volume-snapshot-modal--input"
+          label="Keep"
+          fieldId="snapshot-modal__count"
+        >
+          <div className="ceph-volume-snapshot-modal--input_keep">
+            <Dropdown
+              items={numberOfSnapshot}
+              selectedKey={initialCountKey}
+              onChange={(value) => setSchedulerCountHandler(value)}
+              className="ceph-volume-snapshot-modal--count"
+            />
+            <span className="ceph-volume-snapshot-modal--span_keep">last snapshots</span>
+          </div>
+        </FormGroup>
+        <span className="help-block">*Older snapshots will be deleted automatically</span>
+      </div>
+    );
+  };
+
+  const makeSchedule = (scheduleTemplate: K8sResourceKind) => {
+    handlePromise(k8sCreate(SnapshotScheduleModel, scheduleTemplate))
+      .then(close)
+      .catch((error) => {
+        throw error;
+      });
+  };
   const submit = (event: React.FormEvent<EventTarget>) => {
     event.preventDefault();
     const ns = getNamespace(resource);
     const pvcName = getName(resource);
+    if (scheduleType === 'CronJob') {
+      const pvcLabel = Math.random()
+        .toString(36)
+        .slice(2);
+      const scheduleTemplate: K8sResourceKind = {
+        apiVersion: `${SnapshotScheduleModel.apiGroup}/${SnapshotScheduleModel.apiVersion}`,
+        kind: SnapshotScheduleModel.kind,
+        metadata: {
+          name: snapshotName,
+          namespace: ns,
+        },
+        spec: {
+          claimSelector: {
+            matchLabels: {
+              thislabel: `schedule-${pvcLabel}`,
+            },
+          },
+          disabled: false,
+          retention: {
+            maxCount: snapshotCount,
+          },
+          schedule: scheduleLabel,
+          snapshotTemplate: {
+            snapshotClassName: 'csi-hostpath-snapclass',
+            // TBF - https://github.com/backube/snapscheduler/issues/80
+          },
+        },
+      };
+      const patch = [
+        {
+          op: 'add',
+          path: '/metadata/labels/thislabel',
+          value: `schedule-${pvcLabel}`,
+        },
+      ];
+      handlePromise(k8sPatch(PersistentVolumeClaimModel, resource, patch))
+        .then(() => {
+          makeSchedule(scheduleTemplate);
+        })
+        .catch((error) => {
+          throw error;
```

Comment:
I see no point in catching an error just to throw it again. Not catching it is similar to not throwing it. Maybe `console.error` or just not catch the error.